### DISCUSSION TITLE
Using NEXTCLOUD_DATADIR=/mnt/ncdata should put all sub-containers data within this folder

### DISCUSSION THREAD

**Discussion by nodje** [nextcloud/all-in-one#3575] [Ideas] [answered]
AFAIK, activating `- NEXTCLOUD_DATADIR=/mnt/ncdata` doesn't make the sub-containers put their data in this directory.

I realize each sub-container of AIO has created it's own volume, which is not under control.

I'm not sure what part is stateful and should be backeup., but it seems the database itself has a volume which isn't placed in NEXTCLOUD_DATADIR.

Is there a way to specify where the container data should be stored?

**Comment by Zoey2936** [comment, marked as answer]
see https://github.com/nextcloud/all-in-one#how-to-store-the-filesinstallation-on-a-separate-drive

**Comment by nodje** [comment]
Thanks for your anwser.

I don't think this would be a proper way to manage the storage.
Plus if your docker instance host many different things, this wouldn't be very flexible.

One should be able to have all data related to an app in a single location ideally.
So that you just have to backup this location, and then you have the freedom to wipe the stack and rebuild anywhere.

If AIO claims to be a model deployment for Nextcloud it should offer enough configuration flexibility so that any hosting stack can deploy it.

**Reply by szaimen** [reply]
Hi, if you need this kind of flexibility, see https://github.com/nextcloud/all-in-one/tree/main/manual-install
